### PR TITLE
[6X] Fix segmentation fault during dispatch interrupt

### PR DESCRIPTION
**Pull request by interma**
This PR ported https://github.com/greenplum-db/gpdb/pull/16141 to 6x.

Note: random pipeline failure happened when I merged https://github.com/greenplum-db/gpdb/pull/16299.
After investigation, it was caused by SIGUSR, fix it in this PR (in the second commit).

## Here are some reminders before you submit the pull request
- [x] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [x] Pass `make installcheck`
- [x] Review a PR in return to support the community

### CODE REVIEW

**Comment by interma** [comment]
Dev pipeline passed:
https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-backport16141_6x-rhel8?group=all

**Review by adam8157 — APPROVED** [review]
Do you know who sends the SIGUSR1?

**Comment by interma** [comment]
> Do you know who sends the SIGUSR1?

No, it's little hard to debug it in pipeline env.
But almost confirmed it's from GP rather than external side.

File: src/backend/cdb/dispatcher/test/cdbdisp_query_test.c [hunk 1]
```diff
@@ -284,6 +284,9 @@ test__CdbDispatchPlan_may_be_interrupted(void **state)
 int
 main(int argc, char *argv[])
 {
+	pqsignal(SIGUSR1, SIG_IGN);
```
Comment (Aegeaner) [inline src/backend/cdb/dispatcher/test/cdbdisp_query_test.c:287]: `pqsignal(SIGUSR1, procsignal_sigusr1_handler);` seems better.

**Pull request merged**